apps/vaults/components/SettingsPopover.tsx

```diff
@@ -1,19 +1,31 @@
 import {Fragment, useMemo} from 'react';
 import {Popover, Transition} from '@headlessui/react';
 import {isSolverDisabled} from '@vaults/contexts/useSolver';
+import {useStakingRewards} from '@vaults/contexts/useStakingRewards';
 import {Renderable} from '@yearn-finance/web-lib/components/Renderable';
 import {IconSettings} from '@yearn-finance/web-lib/icons/IconSettings';
+import {Switch} from '@common/components/Switch';
 import {useYearn} from '@common/contexts/useYearn';
 import {Solver} from '@common/schemas/yDaemonTokenListBalances';
 
 import type {ReactElement} from 'react';
 import type {TSolver} from '@common/schemas/yDaemonTokenListBalances';
+import type {TYDaemonVault} from '@common/schemas/yDaemonVaultsSchemas';
 
 type TSettingPopover = {
-	chainID: number
+	vault: TYDaemonVault
```

Comment:
We need already 2 things from the current vault in this settings popover, so just passing the whole vault -- happy to refactor to just pass the `chainID`/`address` if strong opinions against this.